CSharp/Library/QnAMaker/QnAMakerDialog.cs

```diff
@@ -1,110 +1,225 @@
+
+            if (qnaMakerResults != null)
+            {
+                bool match = false;
+                foreach (var qnaMakerResult in qnaMakerResults.Answers)
+                {
+                    if (qnaMakerResult.Questions[0].Equals(selection, StringComparison.OrdinalIgnoreCase))
+                    {
+                        context.PostAsync(qnaMakerResult.Answer);
+                        match = true;
+
+                        var feedbackRecord = default(FeedbackRecord);
+                        context.PrivateConversationData.TryGetValue("feedbackRecord", out feedbackRecord);
+
+                        if (feedbackRecord != null)
+                        {
+                            feedbackRecord.KbQuestion = qnaMakerResult.Questions.FirstOrDefault();
+                            feedbackRecord.KbAnswer = qnaMakerResult.Answer;
+
+                            var tasks =
+                                this.services.Select(
+                                    s =>
+                                    s.ActiveLearnAsync(
+                                        feedbackRecord.UserId,
+                                        feedbackRecord.UserQuestion,
+                                        feedbackRecord.KbQuestion,
+                                        feedbackRecord.KbAnswer)).ToArray();
+                            break;
+                        }
+                    }
+                }
+                if (!match)
+                {
+                    context.PostAsync(
+                        "Not able to match. Please click on the options or type in the exact text from the options.");
+                }
+            }
+            await this.DefaultWaitNextMessageAsync(context, context.Activity.AsMessageActivity(), qnaMakerResults);
+        }
+
+        protected virtual async Task QnAFeedbackStepAsync(IDialogContext context, QnAMakerResults qnaMakerResults)
+        {
+            var qnaList = qnaMakerResults.Answers;
+            var questions = qnaList.Select(x => HttpUtility.HtmlDecode(x.Questions[0])).ToArray();
+
+            PromptDialog.Choice(
+                                context: context,
+                                resume: ResumeAndPostAnswer,
+                                options: questions,
+                                prompt: "I've found multiple responses matching your query. Please select from the following:",
+                                retry: "Please retry!! Click on the options or type in the exact text from the options.");
+        }
+
+        protected virtual async Task RespondFromQnAMakerResultAsync(IDialogContext context, IMessageActivity message, QnAMakerResults result)
+        {
+            await context.PostAsync(result.Answers.FirstOrDefault().Answer);
+        }
+
+        protected virtual async Task DefaultWaitNextMessageAsync(IDialogContext context, IMessageActivity message, QnAMakerResults result)
+        {
+            context.Done(true);

```

Comment:
Yes, that's what we intend to do. We've received few feedback around that from developers since many of them want to use QnA Maker along with some other Dialogs.